### ISSUE TITLE
SpaceAroundOperator Rule does not trigger when expressions are not assigned to a variable

### ISSUE THREAD

**Issue by sstern6** [rrrene/credo#199]
Credo Version: 0.4.12

Given the example:

```
defmodule CredoTests do
  def bar do
    2+4
    2 + 4
  end
end
```

The rule does not fire unless one of the two lines is assigned to a variable.  Can you confirm this is intended functionality.  Though seems like a bug to me.

Ex when rule fires:

```
defmodule CredoTests do
  def bar do
    2+4
    x = 2 + 4
  end
end
```

**Comment by rrrene** [comment]
This is a bug 👍

**Comment by sstern6** [comment]
@rrrene working on this! Should have a PR open soon.

**Comment by sstern6** [comment]
Edit/Update:  Noticed that `3 * 2-3` gets flagged which led me to some discoveries with edge cases like `2+2`, `3 - 2`, '3*2', etc.

Thought about adding logic checking if there was an equal operator in the `line` to `&create_issues?/3` would fix the issue, it does but then it breaks other tests where i didnt account for values such as `@test -1` and 4 other test cases.

Though, I feel like adding another patten match/more logic to the space_around_operators.ex file is making it more fragile and patching one problem using String analysis, which can get sketchy.  Not solving the bigger issue at hand.  Any recommendations @rrrene of how to proceed?

Thanks

**Comment by rrrene** [comment]
The big question here seems to be: Why does it work when there's the assignment? Understanding what goes wrong now will enable us to make it less fragile tomorrow ;-)

**Comment by sstern6** [comment]
@rrrene understood 😄.  What seems to be the issue is the pattern match against https://github.com/rrrene/credo/blob/master/lib/credo/check/consistency/space_around_operators.ex#L84.

Currently there are 2, one for operators `:+, :-`, if thats not the operator coming in, i.e `:=`, then it defers to the `create_issues?` function that returns true, which creates the issue. 

In addition, the way i see it, this logic https://github.com/rrrene/credo/blob/master/lib/credo/check/consistency/space_around_operators.ex#L80-L82 is causing expressions like `2 + 2` to not be in the consistency check because the logic, `!number_with_sign?(line, column)` will return false, not creating the issue for the check.

My opinion is to maintain the case where credo doesnt check values like `x = -1`|| `@x -1`, but I think there should be a case statement/conditional/pattern match that doesnt rely on creating or not creating the issue for all cases based on `!number_with_sign?(line, column)`.  So it seems like we can separate the logic into multiple segments from https://github.com/rrrene/credo/blob/master/lib/credo/check/consistency/space_around_operators.ex#L81.  Not sure how yet but will need to keep digging, also the logic is throwing consistency errors inconsitstently.  For example, just found out,
`3 * 2-3` -> Throws an error.  
`2 - 2*3` -> Does NOT error.

Thanks
